variables.tf

```diff
@@ -1,11 +1,154 @@
+  type        = list(string)
+  description = "List of tags associated with the Event Steams instance"
+  default     = []
+}
+
+variable "region" {
+  type        = string
+  description = "IBM Cloud region where event streams will be created"
+  default     = "us-south"
+}
+
+variable "throughput" {
+  type        = number
+  description = "Throughput capacity in MB per second."
+  default     = "150" # For lite and standard plan, the allowed value of throughput is 150 MB per second. For enterprise plan it can take any value.
+}
+
+variable "storage_size" {
+  type        = number
+  description = "Storage size of the event streams in GB."
+  # For lite and standard plan, the allowed value is 2048 GB. For enterprise plan refer (https://cloud.ibm.com/docs/EventStreams?topic=EventStreams-ES_scaling_capacity#ES_storage_capacity) for allowed values.
+  # Storage capacity cannot be scaled down once instance is created.
+  # Storage capacity cannot be scaled up in lite and standard plan.
+  default = null
+}
+
+variable "service_endpoints" {
+  type        = string
+  description = "The type of service endpoint(public,private or public-and-private) to be used for connection."
+  default     = null
+  validation {
+    condition     = contains(["public", "public-and-private", "private"], var.service_endpoints)
+    error_message = "The specified service endpoint is not a valid selection! Supported options are: public, public-and-private or private."
+  }
+}
+
+variable "private_ip_allowlist" {
+  type        = string
+  description = "Range of IPs that have the access."
+  default     = null
+}
+
+variable "schemas" {
+  type = list(object(
+    {
+      schema_id = string
+      schema = object({
+        type = string
+        name = string
+      })
+    }
+  ))
+  description = "The list of schema object which contains schema id and format of the schema"
+  default     = []
+}
+
+variable "topic_names" {
+  type        = list(string)
+  description = "The name of the topic instance"
+  default     = ["topic_1", "topic_2", "topic_3"] # For lite plan only one topic is allowed
+  validation {
+    condition = alltrue([
+      for name in var.topic_names : length(name) <= 200
+    ])
+    error_message = "Topic name should not be more than 200 characters long"
+  }
+}
+
+variable "partitions" {
+  type        = list(number)
+  description = "The number of partitions in which the topics are to be divided"
+  default     = [1, 1, 1] # For lite plan only one partition per topic is allowed. For the standard plan partition value should not exceed 100 per topic. For enterprise plan it can be set to any value depending upon the capacity.
+}
+
+variable "segment_bytes" {
+  type        = list(number)
+  description = "The maximum size of a partition in bytes"
+  default     = null # For standard plan the range of allowed value is [100 KiB,500 MiB] and for enterprise plan the range of allowed value is [100KiB, 2 TiB]
+}
+
+variable "cleanup_policy" {
```

Comment:
move this variable into a single map variable for topics (see slack thread discussion)